libindy/src/commands/signus.rs

```diff
@@ -477,14 +488,27 @@ impl SignusCommandExecutor {
 
     fn get_endpoint_for_did(&self,
                             wallet_handle: i32,
-                            did: String) -> Result<(String, String), IndyError> {
-        self.signus_service.validate_did(&did)?;
-
-        // TODO: FIXME: It should support resolving of endpoint from ledget!!!
-        let endpoint = self._wallet_get_did_endpoint(wallet_handle, &did)?;
+                            pool_handle: i32,
+                            did: String,
+                            cb: Box<Fn(Result<(String, String), IndyError>) + Send>) {
+        try_cb!(self.signus_service.validate_did(&did), cb);
 
-        let res = (endpoint.ha, endpoint.verkey);
-        Ok(res)
+        check_wallet_and_pool_handles_consistency!(self.wallet_service, self.pool_service,
```

Comment:
pool_handle should be optional. This check should be performed only in case of missed local value.

Reply:
also it will simplify some tests